gufe/solventcomponent.py

```diff
@@ -0,0 +1,88 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/gufe
+from typing import Tuple
+from openff.units import unit
+
+from gufe import Component
+
+
+# really wanted to make this a dataclass but then can't sort & strip ion input
+class SolventComponent(Component):
+
+    """Solvent molecules in a chemical state
+
+    This component represents the abstract idea of the solvent and ions present
+    around the other components, rather than a list of specific water molecules
+    and their coordinates.  This abstract representation is later made concrete
+    by specific methods.
+    """
+    def __init__(self, smiles: str = 'O',
+                 ions: Tuple[str, ...] = None,
+                 neutralize: bool = True,
+                 concentration: unit.Quantity = None):
+        """
+        Parameters
+        ----------
+        smiles : str, optional
+          smiles of the solvent, default 'O' (water)
+        ions : list of str, optional
+          ions in the system, default `None`
```

Comment:
Is this a list of all individual ions, or is this expected to be one positive and one negative? (If so, maybe have `position_ion` and `negative_ion`?) Taking this as a list is confusing me, because I'm trying to figure out what list length 1 or list length 3 might mean.